sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/_package/model_package.py

```diff
@@ -0,0 +1,292 @@
+    ) -> "PackageInputPathVersion":
+        return PackageInputPathVersion(
+            input_path_type=package_input_path_version_rest_object.input_path_type,
+            resource_name=package_input_path_version_rest_object.resource_name,
+            resource_version=package_input_path_version_rest_object.resource_version,
+        )
+
+
+@experimental
+class PackageInputPathUrl:
+    def __init__(self, input_path_type: Optional[str] = None, url: Optional[str] = None):
+        """PackageInputPathUrl.
+
+        :param input_path_type: The type of the input path.
+        :type input_path_type: str
+        :param url: The url of the input path.
+        :type url: str
+        """
+        self.input_path_type = input_path_type
+        self.url = url
+
+    def _to_rest_object(self) -> RestPackageInputPathUrl:
+        return RestPackageInputPathUrl(
+            input_path_type=self.input_path_type,
+            url=self.url,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, package_input_path_url_rest_object: RestPackageInputPathUrl) -> "PackageInputPathUrl":
+        return PackageInputPathUrl(
+            input_path_type=package_input_path_url_rest_object.input_path_type,
+            url=package_input_path_url_rest_object.url,
+        )
+
+
+@experimental
+class ModelPackageInput:
+    """Model package input.
+
+    :param type: The type of the input.
+    :type type: str
+    :param path: The path of the input.
+    :type path: azure.ai.ml.entities.PackageInputPathId
+     or azure.ai.ml.entities.PackageInputPathUrl or azure.ai.ml.entities.PackageInputPathVersion
+    :param mode: The mode of the input.
+    :type mode: str
+    :param mount_path: The mount path of the input.
+    :type mount_path: str
+    """
+
+    def __init__(
+        self,
+        type: Optional[str] = None,
+        path: Optional[Union[PackageInputPathId, PackageInputPathUrl, PackageInputPathVersion]] = None,
+        mode: Optional[str] = None,
+        mount_path: Optional[str] = None,
+    ):
+        self.type = type
+        self.path = path
+        self.mode = mode
+        self.mount_path = mount_path
+
+    def _to_rest_object(self) -> RestModelPackageInput:
+        return RestModelPackageInput(
+            input_type=snake_to_pascal(self.type),
+            path=self.path._to_rest_object(),
+            mode=snake_to_pascal(self.mode),
+            mount_path=self.mount_path,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, model_package_input_rest_object: RestModelPackageInput) -> "ModelPackageInput":
+        return ModelPackageInput(
+            type=model_package_input_rest_object.input_type,
+            path=model_package_input_rest_object.path._from_rest_object(),
+            mode=model_package_input_rest_object.mode,
+            mount_path=model_package_input_rest_object.mount_path,
+        )
+
+
+@experimental
+class ModelPackage(Resource, PackageRequest):
```

Comment:
Are tests in a follow up?

Reply:
Yes tests will be there in the corresponding CLI PR